`indent`: bad indentation in chained methods calls with generics

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

when chaining methods, if one of the method contains generics, the indentation computing fails
In below case it is the reducer function: its callback ask indentation of 2 instead of indentation of 4

```
const map: Record<string, string> = {};

const map2 = Object.keys(map)
  .filter((key) => true)
  .reduce<Record<string, string>>((result, key) => {
    result[key] = map[key];
    return result;
  }, {});
```

```
  6:1  error  Expected indentation of 2 spaces but found 4  @stylistic/indent
  7:1  error  Expected indentation of 2 spaces but found 4  @stylistic/indent
  8:1  error  Expected indentation of 0 spaces but found 2  @stylistic/indent

✖ 3 problems (3 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.
```

### Reproduction

https://stackblitz.com/edit/github-caheqy?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

--- a/packages/eslint-plugin-js/rules/indent/indent.ts
+++ b/packages/eslint-plugin-js/rules/indent/indent.ts
@@ -935,7 +935,7 @@ export default createRule<MessageIds, RuleOptions>({
 
       const offsetAfterToken = node.callee.type === 'TaggedTemplateExpression'
         ? sourceCode.getFirstToken(node.callee.quasi)!
-        : openingParen
+        : node.typeArguments ?? openingParen
       const offsetToken = sourceCode.getTokenBefore(offsetAfterToken)!
 
       offsets.setDesiredOffset(openingParen, offsetToken, 0)

--- a/packages/eslint-plugin-ts/rules/indent/indent.test.ts
+++ b/packages/eslint-plugin-ts/rules/indent/indent.test.ts
@@ -12,6 +12,7 @@ function nonTsTestCase(example: TemplateStringsArray): string {
   return ['// Non-TS Test Case', example].join('\n')
 }
 
+// #region individualNodeTests
 const individualNodeTests = [
   {
     node: AST_NODE_TYPES.ClassDeclaration,
@@ -642,6 +643,7 @@ type Foo = string | {
   },
   { valid: [], invalid: [] },
 )
+// #endregion
 
 run({
   name: 'indent',
@@ -762,6 +764,19 @@ export class Foo {
 
   @a @b() username: string;
 }
+      `,
+      options: [2],
+    },
+
+    // https://github.com/eslint-stylistic/eslint-stylistic/issues/270
+    {
+      code: `
+const map2 = Object.keys(map)
+  .filter((key) => true)
+  .reduce<Record<string, string>>((result, key) => {
+    result[key] = map[key];
+    return result;
+  }, {});
       `,
       options: [2],
     },

